enhancements/network/in-cluster-dns-and-loadbalancers-on-more-platforms.md

```diff
@@ -0,0 +1,297 @@
+cloud environment that's not already supported by the OpenShift installer,
+utilizing the native offerings requires additional work up-front, creates an
+ongoing maintenance burden, and more monetary cost for infrastructure. And not
+all cloud environments offer suitable options. On-prem, there may not exist
+sufficient DNS and/or load balancer services nor additional infrastructure on
+which to run them.
+
+Many users end up deploying a cluster with platform type `baremetal` when all
+they really want is to use the in-cluster network services. Instead, it should
+be possible to utilize those in-cluster network services without them being
+coupled to the baremetal platform type.
+
+For example, the assisted-installer and the agent based installer are often
+used to deploy clusters into “generic” environments where there is not an
+opportunity to utilize an external DNS or load balancer solution. Thus it sets
+the platform type as `baremetal` regardless of whether the systems are actually
+running on bare metal or whether there is any intent to use metal3
+integrations. The resulting cluster has all of the appearances of being bare
+metal, including a BareMetalHost resource for each Node, which can be confusing
+to users. Even the web console’s Overview landing page shows “Infrastructure
+provider: BareMetal” in addition to “N Nodes” and “N Bare Metal Hosts”.
+
+### User Stories
+
+As a user deploying OpenShift in an environment that lacks a suitable DNS
+and/or load balancer solution, and with no intent to utilize metal3-related
+bare metal features, I want to utilize the in-cluster network services without
+being forced to use the `baremetal` platform type.
+
+As a user deploying OpenShift with the `external` platform type into an
+environment of my choosing, I want the option to use the in-cluster network
+services because they are easier to use than manually deploying, configuring
+and managing the alternatives that may be natively available in the
+environment.
+
+As a user deploying OpenShift in a mixed environment, such as [virtualized
+control plane nodes and bare metal worker
+nodes](https://access.redhat.com/solutions/5376701), I am forced to select
+platform type `none`, but I still want the option to use the in-cluster network
+services.
+
+As a developer enabling OpenShift on a new platform via the `external` platform
+type, I want to get an OpenShift cluster up and running with as little friction
+as possible so I can start adding integrations with features of the
+environment.
+
+### Goals
+
+Enable multi-node OpenShift clusters to be viable out-of-the-box in use cases
+where the environment A) lacks a suitable external DNS and/or load balancer
+solution, and B) is not one of the platform types that already provides those
+services in-cluster (`baremetal`, `openstack`, an `ovirt`).
+
+Stop requiring users to select the `baremetal` platform type when all they
```

Comment:
Allow users to opt-in for in-cluster DNS and load balancer services independently of platform type.

Reply:
The proposal as-is doesn't do that. It's just adding the option to two additional platform types.

The alternative described at the bottom would make it independent of platform type, but I don't think that's actually a goal.

WDYT?